### ISSUE TITLE
Empty playlist has an empty source

### ISSUE THREAD

**Issue by danielr18** [benwiley4000/cassette#357]
![image](https://user-images.githubusercontent.com/9046616/52534178-ecff4f80-2d1c-11e9-9057-5df520fac9cc.png)

This triggers `onTrackPlaybackFailure`.  This happens because when playlist is invalid, `const blankSources = [{ src: '' }];` is returned.

I'm happy to make a PR to fix this, what do you think is the best way to handle it?

**Comment by benwiley4000** [comment]
Thanks for catching this!

The "most correct" solution might be to make `blankSources` an empty array, but then we'd have to make a bunch of changes in the rest of the code, so I think the most straightforward solution is to have this check in [the loop that adds source elements](https://github.com/benwiley4000/cassette/blob/next/packages/core/src/PlayerContextProvider.js#L505):

```js
      if (!source.src) {
        continue;
      }
```

**Comment by benwiley4000** [comment]
On the other hand *maybe* we do want to emit an error if someone accidentally adds a track without a source? So not sure if what I said above is ideal. What do you think?

**Comment by danielr18** [comment]
We could check for `isPlaylistValid` before calling `setMediaElementSources` or do something like this in `setMediaElementSources`.
```javascript
if (sources === blankSources) {
  return
}
```

**Comment by benwiley4000** [comment]
> We could check for `isPlaylistValid` before calling `setMediaElementSources`

I like where you're going with this - that makes sense to me.

I think we still want to empty the previous sources and call `media.load()` every time, so maybe we should:

1. Wrap the `for (const source of sources)` loop in a check for `isPlaylistValid`
2. While we're at it, remove the `sources` argument to `setMediaElementSources` and just look it up after the `isPlaylistValid` check for the loop.

**Comment by benwiley4000** [comment]
Closed in #359.
